webdav/webdav_test.go

```diff
@@ -347,3 +370,64 @@ func TestFilenameEscape(t *testing.T) {
 		}
 	}
 }
+
+func TestMoveLockedSrcUnlockedDst(t *testing.T) {
+	// This test reproduces https://github.com/golang/go/issues/43556
+	do := func(method, urlStr string, body string, wantStatusCode int, headers ...string) (http.Header, error) {
+		var bodyReader io.Reader
+		if body != "" {
+			bodyReader = strings.NewReader(body)
+		}
+		req, err := http.NewRequest(method, urlStr, bodyReader)
+		if err != nil {
+			return nil, err
+		}
+		for len(headers) >= 2 {
+			req.Header.Add(headers[0], headers[1])
+			headers = headers[2:]
+		}
+		res, err := http.DefaultTransport.RoundTrip(req)
+		if err != nil {
+			return nil, err
+		}
+		defer res.Body.Close()
+		if res.StatusCode != wantStatusCode {
+			return nil, fmt.Errorf("got status code %d, want %d", res.StatusCode, wantStatusCode)
+		}
+		return res.Header, nil
+	}
+
+	srv := httptest.NewServer(&Handler{
+		FileSystem: NewMemFS(),
+		LockSystem: NewMemLS(),
+	})
+	defer srv.Close()
+
+	src, err := url.Parse(srv.URL)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	src.Path = "/locked_src"
+	hdrs, err := do("LOCK", src.String(), createLockBody, http.StatusCreated)
```

Comment:
*sigh* this will fail to compile. I originally had this branched off another fix where I had pulled the lock body out. I'll fix.